Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,382 @@
+            
+            guard let configuration = makeTraceConfiguration(category: selectedCategory) else {
+                statusText = "Failed to get trace configuration."
+                return
+            }
+            traceParameters.traceConfiguration = configuration
+            
+            let traceResults = try await utilityNetwork
+                .trace(using: traceParameters)
+                .compactMap { $0 as? UtilityElementTraceResult }
+            traceEnabled = false
+            traceCompleted = true
+            if !hasFilterBarriers, let selectedCategory {
+                statusText = "Trace with \(selectedCategory.name.lowercased()) category completed."
+            } else {
+                statusText = "Trace with filter barriers completed."
+            }
+            tracingActivity = .none
+            
+            for result in traceResults {
+                let groups = Dictionary(grouping: result.elements, by: \.networkSource.name)
+                if groups.isEmpty {
+                    statusText = "Trace completed with no output."
+                }
+                for (networkName, elements) in groups {
+                    guard let layer = map.operationalLayers.first(
+                        where: { ($0 as? FeatureLayer)?.featureTable?.tableName == networkName }
+                    ) as? FeatureLayer else { continue }
+                    let features = try await utilityNetwork.features(for: elements)
+                    layer.selectFeatures(features)
+                }
+            }
+            resetEnabled = true
+        }
+        
+        /// Resets the state values for when a trace is cancelled or completed.
+        func reset() {
+            guard traceCompleted else { return }
+            // Reset the trace if it is already completed.
+            layers.forEach { $0.clearSelection() }
+            traceParameters.removeAllBarriers()
+            hasFilterBarriers = false
+            parametersOverlay.removeAllGraphics()
+            traceCompleted = false
+            traceEnabled = true
+            resetEnabled = false
+            selectedCategory = nil
+            // Add back the starting location.
+            addGraphic(for: startingLocationPoint, traceLocationType: "starting point")
+            statusText = "Tap on the map to add filter barriers, or run the trace directly without filter barriers."
+        }
+        
+        /// Get the utility tier's trace configuration and apply category comparison.
+        private func makeTraceConfiguration(category: UtilityCategory?) -> UtilityTraceConfiguration? {
+            // Get a default trace configuration from a tier in the network.
+            guard let configuration = utilityNetwork
+                .definition?
+                .domainNetwork(named: "Pipeline")?
+                .tier(named: "Pipe Distribution System")?
+                .defaultTraceConfiguration
+            else { return nil }
```

Comment:
This should never happen right? If it does then the data changed unexpectedly.  To me that seems like it should fatal error if it does return nil in that case? Making the return value optional has downstream effects.